src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingReconciliationService.kt

```diff
@@ -31,35 +31,52 @@ class SentencingReconciliationService(
-    val activePrisoners = getActivePrisonersForPage(allPrisoners, page)
+  suspend fun generateReconciliationReport(allPrisoners: Boolean): List<MismatchSentencingAdjustments> {
+    val mismatches: MutableList<MismatchSentencingAdjustments> = mutableListOf()
+    var lastBookingId = 0L
 
-    withContext(Dispatchers.Unconfined) {
-      activePrisoners.map { async { checkBookingAdjustmentsMatch(it) } }
+    do {
+      val page = getNextBookingsForPage(lastBookingId, allPrisoners)
+      page?.takeIf { it.prisonerIds.isNotEmpty() }?.also {
+        it.checkPageOfBookingAdjustmentsMatch().also { result ->
+          mismatches += result.mismatches
+          lastBookingId = result.lastBookingId
+        }
+      } ?: run {
+        // just skip this "page" by moving the bookingId pointer up
+        lastBookingId += pageSize.toInt()
+      }
+    } while (page.notLastPage())
+
+    return mismatches.toList()
+  }
+
+  private suspend fun BookingIdsWithLast.checkPageOfBookingAdjustmentsMatch(): MismatchPageResult {
+    val prisonerIds = this.prisonerIds
+    val mismatches = withContext(Dispatchers.Unconfined) {
+      prisonerIds.map { async { checkBookingAdjustmentsMatch(it) } }
     }.awaitAll().filterNotNull()
+    return MismatchPageResult(mismatches, prisonerIds.last().bookingId)
   }
 
-  // TODO - feature switch getActivePrisoners get all prisoners
-  private suspend fun getActivePrisonersForPage(@Suppress("UNUSED_PARAMETER") allPrisoners: Boolean, page: Pair<Long, Long>) =
-    runCatching {
-      doApiCallWithRetries {
-        nomisApiService.getActivePrisoners(
-          pageNumber = page.first,
-          pageSize = page.second,
-        )
-      }.content
-    }
+  data class MismatchPageResult(val mismatches: List<MismatchSentencingAdjustments>, val lastBookingId: Long)
+
+  // Last page will be a non-null page with items less than page size
+  private fun BookingIdsWithLast?.notLastPage(): Boolean = this == null || this.prisonerIds.size == pageSize.toInt()
+
+  private suspend fun getNextBookingsForPage(lastBookingId: Long, allPrisoners: Boolean): BookingIdsWithLast? =
+    runCatching { nomisApiService.getAllLatestBookings(lastBookingId = lastBookingId, activeOnly = !allPrisoners, pageSize = pageSize.toInt()) }
       .onFailure {
         telemetryClient.trackEvent(
           "sentencing-reports-reconciliation-mismatch-page-error",
           mapOf(
-            "page" to page.first.toString(),
+            "booking" to lastBookingId.toString(),
           ),
         )
-        log.error("Unable to match entire page of prisoners: $page", it)
+        log.error("Unable to match entire page of bookings from booking: $lastBookingId", it)
       }
-      .getOrElse { emptyList() }
-      .also { log.info("Page requested: $page, with ${it.size} active prisoners") }
+      .getOrNull()
```

Comment:
we need a way of indicating that this call failed so just continue with next page - using null to indicate that.

I might re-think this - it is a bit shoddy